app/src/main/java/ch/epfl/skysync/ui/components/FlightForm.kt

```diff
@@ -0,0 +1,666 @@
+                              expanded = expandedVehicleMenu,
+                              onDismissRequest = { expandedVehicleMenu = false }) {
+                                allVehicles.withIndex().forEach { (id, item) ->
+                                  if (!listVehiclesValue.contains(item)) {
+                                    DropdownMenuItem(
+                                        modifier =
+                                            Modifier.fillMaxWidth()
+                                                .padding(defaultPadding)
+                                                .testTag("Vehicle $id"),
+                                        onClick = {
+                                          listVehiclesValue.add(item)
+                                          expandedVehicleMenu = false
+                                          addVehicle = false
+                                        },
+                                        text = { Text(item.name) })
+                                  }
+                                }
+                              }
+                        }
+                  }
+                }
+                // Drop down menu for the balloon
+                item {
+                  val balloonTitle = "Balloon"
+                  Text(
+                      modifier = Modifier.padding(horizontal = defaultPadding),
+                      text = balloonTitle,
+                      style = MaterialTheme.typography.headlineSmall)
+                  MenuDropDown(
+                      defaultPadding = defaultPadding,
+                      title = balloonTitle,
+                      value = balloonValue,
+                      onclickMenu = { item -> balloonValue = item },
+                      items = balloons)
+                }
+                // Drop down menu for the basket
+                item {
+                  val basketTitle = "Basket"
+                  Text(
+                      modifier = Modifier.padding(horizontal = defaultPadding),
+                      text = basketTitle,
+                      style = MaterialTheme.typography.headlineSmall)
+                  MenuDropDown(
+                      defaultPadding = defaultPadding,
+                      title = basketTitle,
+                      value = basketValue,
+                      onclickMenu = { item -> basketValue = item },
+                      items = baskets)
+                }
+              }
+          // Button to add the flight to the list of flights
+          Button(
+              modifier = Modifier.fillMaxWidth().padding(defaultPadding).testTag("$title Button"),
+              onClick = {
+                nbPassengersValueError =
+                    nbPassengersValue.isEmpty() || nbPassengersValue.toInt() <= 0
+                flightTypeValueError = flightTypeValue == null
+                isError = nbPassengersValueError || flightTypeValueError
+                if (!isError) {
```

Comment:
Input validation should be extracted in a separated function (for example `inputValidation(): Boolean`)

Reply:
I have put the assignation of the error variables in functions but I don't think it is exactly what you were thinking about

Reply:
I did something with it but I am not sure if it really was what you meant

Reply:
It's already better